UnnecessaryParentheses reports incorrect violation message for return value in lambdas

I have read check documentation: https://checkstyle.org/config_coding.html#UnnecessaryParentheses
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```shell
$ javac ParenthesesTest.java
# ParenthesesTest.class created

$ type config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
        <module name="TreeWalker">
                <module name="UnnecessaryParentheses"/>
        </module>
</module>

$ type ParenthesesTest.java
import java.util.function.Function;

public class ParenthesesTest {
    public void returnValueExample() {
        Function<Integer, Integer> addFive = x -> {
            return (x + 5); // violation, should be reported for parentheses around return value
        };
    }
}

$ java "-Duser.language=en -Duser.country=US" -jar checkstyle-10.0-all.jar -c config.xml ParenthesesTest.java
Starting audit...
[ERROR] C:\Users\Faraaz\CheckstyleTest\ParenthesesTest.java:6:20: Unnecessary parentheses around assignment right-hand side. [UnnecessaryParentheses]
Audit done.
Checkstyle ends with 1 errors.
```
---

When a lambda expression is assigned to a variable and has a return value surrounded in parentheses, the violation reported is ' Unnecessary parentheses around assignment right-hand side.'.

**Expected:**
The correct violation message should be 'Unnecessary parentheses around return value.'.
ParenthesesTest.java:6:20 should be reported for unnecessary parentheses around return value.

---

Could it be due to [this](https://github.com/checkstyle/checkstyle/blob/4981aad0c6a4270f618fcab1b50a00c0c22a33f0/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java#L585), wherein the assignment depth is being checked **before** checking whether the parent token is a 'return'?

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java
@@ -582,11 +582,11 @@ public class UnnecessaryParenthesesCheck extends AbstractCheck {
         // warning about an immediate child node in visitToken, so we don't
         // need to log another one here.
         if (parentToSkip != ast && isExprSurrounded(ast)) {
-            if (assignDepth >= 1) {
-                log(ast, MSG_ASSIGN);
-            }
-            else if (ast.getParent().getType() == TokenTypes.LITERAL_RETURN) {
+            if (ast.getParent().getType() == TokenTypes.LITERAL_RETURN) {
                 log(ast, MSG_RETURN);
+            }
+            else if (assignDepth >= 1) {
+                log(ast, MSG_ASSIGN);
             }
             else {
                 log(ast, MSG_EXPR);

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheckTest.java
@@ -127,19 +127,33 @@ public class UnnecessaryParenthesesCheckTest extends AbstractModuleTestSupport {
     }
 
     @Test
+    public void testReturn() throws Exception {
+        final String[] expected = {
+            "21:33: " + getCheckMessage(MSG_RETURN),
+            "22:16: " + getCheckMessage(MSG_RETURN),
+            "25:16: " + getCheckMessage(MSG_RETURN),
+            "28:16: " + getCheckMessage(MSG_RETURN),
+            "31:16: " + getCheckMessage(MSG_RETURN),
+            "36:16: " + getCheckMessage(MSG_RETURN),
+        };
+        verifyWithInlineConfigParser(
+                getPath("InputUnnecessaryParenthesesReturnValue.java"), expected);
+    }
+
+    @Test
     public void testUnnecessaryParenthesesSwitchExpression() throws Exception {
         final String[] expected = {
             "21:31: " + getCheckMessage(MSG_ASSIGN),
             "24:13: " + getCheckMessage(MSG_LITERAL, 2),
             "25:39: " + getCheckMessage(MSG_ASSIGN),
-            "30:18: " + getCheckMessage(MSG_ASSIGN),
+            "30:18: " + getCheckMessage(MSG_RETURN),
             "32:16: " + getCheckMessage(MSG_IDENT, "g"),
-            "36:18: " + getCheckMessage(MSG_ASSIGN),
+            "36:18: " + getCheckMessage(MSG_RETURN),
             "46:31: " + getCheckMessage(MSG_ASSIGN),
             "48:13: " + getCheckMessage(MSG_LITERAL, 2),
             "49:39: " + getCheckMessage(MSG_ASSIGN),
-            "53:18: " + getCheckMessage(MSG_ASSIGN),
-            "58:18: " + getCheckMessage(MSG_ASSIGN),
+            "53:18: " + getCheckMessage(MSG_RETURN),
+            "58:18: " + getCheckMessage(MSG_RETURN),
         };
         verifyWithInlineConfigParser(
                 getNonCompilablePath(

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java
@@ -27,13 +27,13 @@ public class InputUnnecessaryParenthesesCheckSwitchExpression {
       }
       case 3 -> {
         MathOp2 g = (int a, int b) -> {
-          return (a + b); // violation 'Unnecessary parentheses around assignment right-hand side'
+          return (a + b); // violation 'Unnecessary parentheses around return value'
         };
         yield (g); // violation 'Unnecessary parentheses around identifier 'g''
       }
       default -> {
         MathOp2 h = (int x, int y) -> {
-          return (x + y); // violation 'Unnecessary parentheses around assignment right-hand side'
+          return (x + y); // violation 'Unnecessary parentheses around return value'
         };
         yield h;
       }
@@ -50,12 +50,12 @@ public class InputUnnecessaryParenthesesCheckSwitchExpression {
       }
       case 3 -> {
         MathOp2 g = (int a, int b) -> {
-          return (a + b + 2); // violation 'Unnecessary parentheses around assignment right.*side'
+          return (a + b + 2); // violation 'Unnecessary parentheses around return value'
         };
       }
       default -> {
         MathOp2 h = (int x, int y) -> {
-          return (x + y); // violation 'Unnecessary parentheses around assignment right-hand side'
+          return (x + y); // violation 'Unnecessary parentheses around return value'
         };
       }
     }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesReturnValue.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesReturnValue.java
@@ -1,0 +1,45 @@
+/*
+UnnecessaryParentheses
+tokens = (default)EXPR, IDENT, NUM_DOUBLE, NUM_FLOAT, NUM_INT, NUM_LONG, \
+         STRING_LITERAL, LITERAL_NULL, LITERAL_FALSE, LITERAL_TRUE, ASSIGN, \
+         BAND_ASSIGN, BOR_ASSIGN, , BXOR_ASSIGN, DIV_ASSIGN, \
+         MINUS_ASSIGN, MOD_ASSIGN, PLUS_ASSIGN, SL_ASSIGN, SR_ASSIGN, STAR_ASSIGN, \
+         LAMBDA, TEXT_BLOCK_LITERAL_BEGIN, LAND, LITERAL_INSTANCEOF, GT, LT, GE, \
+         LE, EQUAL, NOT_EQUAL, UNARY_MINUS, UNARY_PLUS, INC, DEC, LNOT, BNOT, \
+         POST_INC, POST_DEC
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
+
+import java.util.function.Function;
+
+public class InputUnnecessaryParenthesesReturnValue {
+    int foo() {
+        Function<Integer, Integer> addOne
+                = x -> { return (x + 1); }; // violation 'Unnecessary paren.* around return'
+        return (1 + 1); // violation 'Unnecessary paren.* around return'
+    }
+    boolean compare() {
+        return (9 <= 3); // violation 'Unnecessary paren.* around return'
+    }
+    boolean bar() {
+        return (true && 7 > 3); // violation 'Unnecessary paren.* around return'
+    }
+    boolean bigger() {
+        return (Integer.parseInt("5") > 7 || // violation 'Unnecessary paren.* around return'
+                Integer.parseInt("2") > 3 ||
+                "null" != null);
+    }
+    int ternary() {
+        return (true ? 0 : 1); // violation 'Unnecessary paren.* around return'
+    }
+    boolean ok() {
+        return 5 > 7 ||
+                6 < 4;
+    }
+    int okternary() {
+        return true ? 0 : 1;
+    }
+}

